File: term1/hw/program_design/frag.py
def helper( n,  m):
    result = 0
    if ( n == 1 or n == 0 ):
        return 1
    elif n < 0:
        return 0
    for i in range(1, m+1):
        result += helper(n-i, m)
    return result

def main():
    n = int(input())
    m = int(input())
    print(helper(n, m))
    return 0

main()

File: term1/hw/program_design/frag.c
# include <stdio.h>

int helper(int n){
    if ( n == 0 || n == 1 ){
        return 1;
    }
    else{
        return helper(n-1) + helper(n-2);
    }

}

int main(void){
    int n = 0;
    scanf("%d", &n);
    printf("%d", helper(n));
    return 0;
}
